scripts/ecs-anywhere-install.sh

```diff
@@ -0,0 +1,526 @@
+    fi
+fi
+
+ok() {
+    echo ""
+    echo "# ok"
+    echo "##########################"
+    echo ""
+}
+
+try() {
+    local action=$*
+    echo ""
+    echo "##########################"
+    echo "# Trying to $action ... "
+    echo ""
+}
+
+PKG_MANAGER=""
+if [ -x "$(command -v dnf)" ]; then
+    PKG_MANAGER="dnf"
+    dnf install -y jq
+elif [ -x "$(command -v yum)" ]; then
+    PKG_MANAGER="yum"
+    yum install epel-release -y
+    yum install -y jq
+elif [ -x "$(command -v apt)" ]; then
+    PKG_MANAGER="apt"
+    try "Update apt repos"
+    apt update -y
+    apt-get install -y curl jq
+    ok
+elif [ -x "$(command -v zypper)" ]; then
+    PKG_MANAGER="zypper"
+    zypper install -y jq
+else
+    echo "Unsupported package manager. Could not find dnf, yum, or apt."
+    fail
+fi
+
+INSTANCE_REGISTERED=false
+check-instance-registered() {
+    try "checking if instance already has an SSM managed instance ID."
+    SSM_REGISTRATION_FILE='/var/lib/amazon/ssm/registration'
+    if [ -f ${SSM_REGISTRATION_FILE} ]; then
+        MANAGED_INSTANCE_ID=$(jq -r ".ManagedInstanceID" $SSM_REGISTRATION_FILE)
+        if [ ! -z $MANAGED_INSTANCE_ID ]; then
+            INSTANCE_REGISTERED=true
+            return
+        fi
+    fi
+    ok
+}
+
+register-ssm-agent() {
+    check-instance-registered
+    if ! $INSTANCE_REGISTERED; then
+        SERVICE_NAME="amazon-ssm-agent"
+        systemctl stop "$SERVICE_NAME"
+        amazon-ssm-agent -register -code "$ACTIVATION_CODE" -id "$ACTIVATION_ID" -region "$REGION"
+        systemctl enable "$SERVICE_NAME"
+        systemctl start "$SERVICE_NAME"
+        echo "Instance is registered."
+    else
+        echo "instance registration found."
+    fi
+}
+
+install-ssm-agent() {
+    try "install ssm agent"
+    if [ "$(systemctl is-enabled snap.amazon-ssm-agent.amazon-ssm-agent.service)" == "enabled" ]; then
+        check-instance-registered
+        if $INSTANCE_REGISTERED; then
+            echo "Instance has been registered."
+            return
+        else
+            echo "We currently don't support instance registration on ssm agent installed via Snap. Please ensure your instance is registered and then rerun the script with --skip-registration flag."
+            fail
+        fi
+    elif [ "$(systemctl is-enabled amazon-ssm-agent)" == "enabled" ]; then
+        echo "ssm agent is installed, checking if the instance is registered."
+        check-instance-registered
+        if $INSTANCE_REGISTERED; then
+            echo "Instance has been registered."
+            return
+        else
+            echo "SSM Agent is installed. The instance needs to be registered."
+            register-ssm-agent
+            return
+        fi
+    else
+        local dir
+        dir="$(mktemp -d)"
+        ssm-agent-signature-verify
+        case "$PKG_MANAGER" in
+        dnf)
+            dnf install -y "https://s3.$REGION.amazonaws.com/amazon-ssm-$REGION/latest/linux_$ARCH_ALT/amazon-ssm-agent.rpm"
```

Comment:
since ssm-agent-signature-verify downloads the rpm already, we should avoid duplicate download and just install the local one, same for yum

Reply:
Updated.